### PR TITLE
Add Timeout & Cancellation Example to `features.md`

### PR DESCRIPTION
**Pull request by justsml**
Hey @benjamingr @spion @petkaantonov and anyone else interested in docs.

I was hoping to add more examples and/or do write ups on the blank pages (i.e. tutorials).

Is there anything in particular you are looking for? (Should a tutorial be something like "HowTo: 'Promise' a Twitter Bootstrap's Dialog" or "HowTo: Replace $q with Bluebird in Angular"? )

Anyway, looking forward to your feedback on this PR. 
Thx!

### CODE REVIEW

**Comment by justsml** [comment]
I saw #900 ... Really good stuff, can't wait for it to get merged. Let me know if I can help it along.

File: docs/docs/features.md [hunk 1]
```diff
@@ -292,6 +292,29 @@ window.onpromisechained = function(promise, child) {
...
+Promise.config({cancellation: true});
+
+var fs = Promise.promisifyAll(require("fs"));
+
+// In 2000ms or less, load & parse a file 'config.json'
+var p = Promise.resolve('./config.json')
+ .timeout(2000)
+ .catch(console.error.bind(console, 'Failed to load config!'))
+ .then(fs.readFileAsync)
+ .then(function(data) {
```
Comment (petkaantonov) [inline docs/docs/features.md]: could be `JSON.parse` passed directly as in `.then(JSON.parse)`

**Comment by benjamingr** [comment]
Wow sorry totally missed this - a beginner's guide would be awesome.

**Comment by justsml** [comment]
Thanks guys, I just noticed this got merged ;)

I've been pulling together a collection of snippets showing how to promise-ify common patterns... 

I'll spend some time next weekend putting together a side-by-side code comparison page...

Let me know if you have any ideas/libraries you want to see...

**Pull request merged**